simpleble/src/backends/macos/PeripheralBaseMacOS.h

```diff
@@ -19,6 +19,7 @@
 - (uint16_t)mtu;
 
 - (void)connect;
+- (void)connectWithTimeout:(NSTimeInterval)timeout;
```

Comment:
There's no need to create another function. `connect` should have a reasonable timeout baked in.